builtins/amp-img.js

```diff
@@ -65,8 +65,21 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
+  preconnectCallback(onLayout) {
+    // NOTE(@wassgha) the correct src url isn't determined until the element is
+    // laid out, thus assuming that hopefully all urls in the srcset/src
+    // belong to the same domain, we take the first url and pre-connect to it.
+    // If the image doesn't have an srcset, this connects to the src url.
+    this.preconnect.url(this.srcset_[0].url, onLayout);
```

Comment:
`this.srcset_` is not an array, it is a `srcset` object (class defined in srcset.js). You probably want 
```
this.preconnect.url(this.srcset_.getLast().url, onLayout);
```

`getLast()` gives you the `default` source which is probably the best bet for preconnect.